### ISSUE TITLE
Support for preview documents

### ISSUE THREAD

**Issue by sarkoziadam** [filegator/filegator#98]
Dear Developers,

Could you please implement an offline document viewer?
For example doc, docx, xls, xlsx may have an embedded pdf viewer.
It would be a very useful feature. (Tiny File Manager has this feature, but unfortunately not offline.)
(Offline means you can't just view it over an internal network, you need wan access.)

**Comment by alcalbg** [comment]
Yes, the other file manager is using google docs to embed and preview files in an iframe. But as you said, if google can't access your files, then this doesn't work.

I don't know if there is a good solution for this at this point...

**Comment by sarkoziadam** [comment]
A pdf conversion solution may be appropriate.
Now it comes to my mind that embedded google docs would also be good because the file names are "encrypted" so it can be safe.
Would you please implement one of the two?

Thanks in advance!

**Comment by alcalbg** [comment]
Ref #141

**Comment by samansupriadi** [comment]
when i try to open doc it said 
File not found
![image](https://github.com/filegator/filegator/assets/12885184/67d074c6-a093-40b1-9ea8-3924c414d91f)


and error log apache like :


var/www/filegator/repository/undefined/Text/notulen


it say undefined. its like filegator can not read for folde home when read doc file.
how can i resolve this ?

**Comment by samansupriadi** [comment]
> Yes, the other file manager is using google docs to embed and preview files in an iframe. But as you said, if google can't access your files, then this doesn't work.
> 
> I don't know if there is a good solution for this.
> 
> One solution would be to convert everything to PDF on the fly and display that to the user. So basically, when user wants to preview `invoice.docx` file, convert that to `tmp/invoice.pdf` and show him. Same goes for .xlsx, .ods, .ppt or any other 'supported' format.

could you give me the step ?
